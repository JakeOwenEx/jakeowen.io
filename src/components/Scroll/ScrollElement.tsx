import * as React from "react";
import { ReactElement } from "react";
import * as Scroll from "react-scroll";

const Element = Scroll.Element;

interface ScrollElement {
  name: string;
  component: ReactElement;
}
export const ScrollElement: React.FC<ScrollElement> = (props) => {
  return <Element name={props.name}>{props.component}</Element>;
};
