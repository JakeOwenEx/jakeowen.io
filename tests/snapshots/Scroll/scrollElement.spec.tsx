import * as React from "react";
import { create } from "react-test-renderer";
import { ScrollElement } from "../../../src/components/Scroll/ScrollElement";
import ObjMock from "../../__mocks__/fileMock";

describe("Snapshot Sections", () => {
  it("Scroll Element", () => {
    expect(
      create(<ScrollElement name={"any-name"} component={ObjMock} />),
    ).toMatchSnapshot();
  });
});
