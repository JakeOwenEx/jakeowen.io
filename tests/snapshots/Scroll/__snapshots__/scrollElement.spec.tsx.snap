// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Sections Scroll Element 1`] = `
<div
  name="any-name"
/>
`;
